pilot/pkg/proxy/envoy/v2/eds.go

```diff
@@ -821,7 +823,12 @@ func (s *DiscoveryServer) pushEds(push *model.PushContext, con *XdsConnection, e
 			// Make a shallow copy of the cla as we are mutating the endpoints with priorities/weights relative to the calling proxy
 			clonedCLA := util.CloneClusterLoadAssignment(l)
 			l = &clonedCLA
-			loadbalancer.ApplyLocalityLBSetting(con.modelNode.Locality, l, s.Env.Mesh.LocalityLbSetting)
+			// TODO(#12961) get outlierDetection from the cluster
```

Comment:
you have the cluster name here. So you can certainly look up the Cluster from cds.rawClusters (I think) and obtain the outlier detection value..

Reply:
I looked at it for a bit and the only thing I could find was using ConfigGenerator.buildClusters. I would be worried about performance here if we build the clusters for each push. If that's not the case that would be great though

Reply:
Another way is to 1. get the destinationrule of the cluster 2. check if the outlierDetection exists

FYI: https://github.com/istio/istio/pull/10720/files#diff-f507fb55b7edf4a03d3c7e9aebf64a1eR156